### PR TITLE
Mongodb kubernetes

### PR DESCRIPTION
**Pull request by raymond-selvik**

### CODE REVIEW

File: infrastructure/docker/docker-compose.yml [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-      - mongodb_data_container:/data/db
-
-volumes:
+version: '3.7'
+services:
+  mongodb_container:
+    image: mongo:4.4
+    environment:
+      MONGO_INITDB_ROOT_USERNAME: root
+      MONGO_INITDB_ROOT_PASSWORD: rootpassword
```
Comment (LERoise) [inline infrastructure/docker/docker-compose.yml:7]: nice password :D

File: infrastructure/kubernetes/mongodb/mongodb-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    matchLabels:
+      app: mongodb
+  template:
+    metadata:
+      labels:
+        app: mongodb
+    spec:
+      containers:
+      - name: mongodb
+        image: mongo
```
Comment (martinothamar) [inline infrastructure/kubernetes/mongodb/mongodb-deployment.yaml:19]: Put a version tag here maybe so it's pinned

**Review by LERoise — APPROVED** [review]

File: infrastructure/kubernetes/mongodb/mongodb-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        ports:
+        - containerPort: 27017
+        env:
+        - name: MONGO_INITDB_ROOT_USERNAME
+          value: root
+        - name: MONGO_INITDB_ROOT_PASSWORD
+          value: password
```
Comment (martinothamar) [inline infrastructure/kubernetes/mongodb/mongodb-deployment.yaml:26]: From a secret perhaps?

**Review by martinothamar — COMMENTED** [review]
Where should the data live? I.e. where's the persistent volume. And have you considered using premade helm chart for this? The deployment and service can also be in a single file so that CI/CD only needs to point to 1 manifest

File: infrastructure/docker/docker-compose.yml (hunk 1 above)
Comment (martinothamar) [reply infrastructure/docker/docker-compose.yml:7]: I may be at fault here, but it was only intended for local use 🥇

Comment (raymond-selvik) [reply infrastructure/docker/docker-compose.yml:7]: It is only intended for local use as for now. I will create a own Secret.yaml in the future, for deployment into AKS and so. And I need to investigate how to integrate with Azure Key Vault, where we should store the secrets.

File: infrastructure/kubernetes/mongodb/mongodb-deployment.yaml (hunk 3 above)
Comment (raymond-selvik) [reply infrastructure/kubernetes/mongodb/mongodb-deployment.yaml:26]: It is only intended for local use as for now. I will create a own Secret.yaml in the future, for deployment into AKS and so. And I need to investigate how to integrate with Azure Key Vault, where we should store the secrets.

**Comment by raymond-selvik** [comment]
Regarding storage, I see there are different options one could use. One is to simple create a volume. But I also see some uses a kind of resource called StatefullSet instead of Deployment. Do someone have experience with that?

**Comment by martinothamar** [comment]
You need pv's and pvc's either way. StatefuleSet's differ from Deployment's in scheduling constraints and how persistent volumes are attached AFAIK, but running stateful workloads in k8s is pretty difficult so I tend to outsource those considerations to helm chart devs (https://github.com/bitnami/charts/tree/master/bitnami/mongodb). But since this was only for local use it doesn't matter much. The docker-compose file uses a volume because you want to restart the mongodb container but still keep the data, and if you  want to reset the data you can do `docker-compose down -v`

**Pull request merged**